src/Parse/Parse.hs

```diff
@@ -159,9 +160,28 @@ parseTerm1 = parsePeeks 2 >>= \ t1t2 -> case t1t2 of
 -- \ x [: type] . term
   [TkLam, _] -> parseEat *> pure UsLam <*> parseVar <*> parseTpAnn <* parseDrop TkDot <*> parseTerm1
 -- let (x, y, ...) = term in term
-  [TkLet, TkParenL] -> parseEat *> parseEat *> pure (flip (UsElimProd Multiplicative)) <*> parseVarsCommas TkParenR True False <* parseDrop TkEq <*> parseTerm1 <* parseDrop TkIn <*> parseTerm1
+  [TkLet, TkParenL] -> do
+    parseEat
+    parseEat
+    xs <- parseVarsCommas TkParenR True False
+    parseDrop TkEq
+    tm <- parseTerm1
+    parseDrop TkIn
+    tm' <- parseTerm1
+    return (UsElimMultiplicative tm xs tm')
 -- let <..., _, x, _, ...> = term in term
-  [TkLet, TkLangle] -> parseEat *> parseEat *> pure (flip (UsElimProd Additive)) <*> parseVarsCommas TkRangle False True <* parseDrop TkEq <*> parseTerm1 <* parseDrop TkIn <*> parseTerm1
+  [TkLet, TkLangle] -> do
+    parseEat
+    parseEat
+    xs <- parseVarsCommas TkRangle False True
+    case [ (i,x) | (i,x) <- enumerate xs, x /= Var "_" ] of
```

Comment:
Why not push the distinction earlier and make "_" a different Token?

Reply:
Oh that's a good point. I think we should continue to allow _ in lambda, case, and ElimMultiplicative though, and I'm not sure the best way to represent that...

Reply:
Do you want _ to have the meaning "ignore me" or "variable named _"?